packages/flutter/test/widgets/snapshot_widget_test.dart

```diff
@@ -0,0 +1,237 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This file is run as part of a reduced test set in CI on Mac and Windows
+// machines.
+@Tags(<String>['reduced-test-set'])
+
+import 'dart:ui' as ui;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('SnapshotWidget can rasterize child', (WidgetTester tester) async {
+    final SnapshotWidgetController controller = SnapshotWidgetController(enabled: true);
+    final Key key = UniqueKey();
+    await tester.pumpWidget(RepaintBoundary(
+      key: key,
+      child: Center(
+        child: SnapshotWidget(
+          controller: controller,
+          child: Container(
+            width: 100,
+            height: 100,
+            color: const Color(0xFFAABB11),
+          ),
+        ),
+      ),
+    ));
+    // Rasterization is not actually complete until a frame has been pumped through
+    // the engine.
+    await tester.pumpAndSettle();
+
+    await expectLater(find.byKey(key), matchesGoldenFile('raster_widget.yellow.png'));
+
+    // Now change the color and assert the old snapshot still matches.
+    await tester.pumpWidget(RepaintBoundary(
+      key: key,
+      child: Center(
+        child: SnapshotWidget(
+          controller: controller,
+          child: Container(
+            width: 100,
+            height: 100,
+            color: const Color(0xFFAA0000),
+          ),
+        ),
+      ),
+    ));
+    await tester.pumpAndSettle();
+
+    await expectLater(find.byKey(key), matchesGoldenFile('raster_widget.yellow.png'));
+
+    // Now invoke clear and the raster is re-generated.
+    controller.clear();
+    await tester.pumpAndSettle();
```

Comment:
`await tester.pump()` should do here? This all is expected to happen in the next frame, no?

Reply:
Done